From 6bca61fe8a1183bf5f457f989a02678ce75374e9 Mon Sep 17 00:00:00 2001
From: Masami Hiramatsu <masami.hiramatsu@linaro.org>
Date: Fri, 16 Apr 2021 14:53:46 -0700
Subject: [PATCH 04/19] pci: Update the highest subordinate bus number for
 bridge setup

Update the highest subordinate bus number after probing the devices
under the bus for setting up the bridge correctly.
The commit ("pci: Update to use new sequence numbers")
removed this but it is required if a PCIe bridge is under the bus.

Fixes: "pci: Update to use new sequence numbers"

Issue: ALB-9211

Upstream-Status: Pending 

Signed-off-by: Masami Hiramatsu <masami.hiramatsu@linaro.org>
Signed-off-by: Ionut Vicovan <Ionut.Vicovan@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/pci/pci-uclass.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/pci/pci-uclass.c b/drivers/pci/pci-uclass.c
index c6081bc822..87dea29c33 100644
--- a/drivers/pci/pci-uclass.c
+++ b/drivers/pci/pci-uclass.c
@@ -630,6 +630,8 @@ int dm_pci_hose_probe_bus(struct udevice *bus)
 		return log_msg_ret("probe", ret);
 	}
 
+	sub_bus = pci_get_bus_max();
+
 	dm_pciauto_postscan_setup_bridge(bus, sub_bus);
 
 	return sub_bus;
-- 
2.17.1

